homeassistant/helpers/entity_platform.py

```diff
@@ -640,7 +641,7 @@ def async_register_entity_service(
             return
 
         if isinstance(schema, dict):
```

Comment:
I honestly think we should not accept `vol.Schema` for this method... all these entity services should accept targets and this can help to ensure that. I've seen too many occurrences of people dropping in a Schema and next wondering why target's don't work.

@MartinHjelmare What do you think?

Reply:
Yeah, I don't see a reason right now why we need to accept a `Schema` except for legacy reasons.

Reply:
ok, I'll adjust to that in that case.

Reply:
It looks like `entity_service_call` in the service helper won't call any entity method unless there's at least one entity identifier, entity_id, device_id, area_id. So it's really bad to let the integrations handle creating the schema.

Reply:
Agreed, however, we also have cases like these:

```py
SET_TEMPERATURE_SCHEMA = vol.All(
    cv.has_at_least_one_key(
        ATTR_TEMPERATURE, ATTR_TARGET_TEMP_HIGH, ATTR_TARGET_TEMP_LOW
    ),
    make_entity_service_schema(
        {
            vol.Exclusive(ATTR_TEMPERATURE, "temperature"): vol.Coerce(float),
            vol.Inclusive(ATTR_TARGET_TEMP_HIGH, "temperature"): vol.Coerce(float),
            vol.Inclusive(ATTR_TARGET_TEMP_LOW, "temperature"): vol.Coerce(float),
            vol.Optional(ATTR_HVAC_MODE): vol.In(HVAC_MODES),
        }
    ),
)
```

(Climate entity-component)

Allowing that makes sense in a way?

Reply:
Yes, I guess we can't avoid those. :sob:

Reply:
Yup, seems like it. Nevertheless, found a bunch of broken entity service schema because I made the change locally for a moment, so will open PRs for those.